### ISSUE TITLE
Publish manual as a website

### ISSUE THREAD

**Issue by nawordar** [sile-typesetter/sile#1692]
I'm just trying SILE and I stumbled upon the same problem as with TeX, LuaTeX, ConTeXt etc. — the documentation is only available as PDF which is not easily searchable. I understand that PDF allows to show off the typesetter capabilities, but it's just not very practical for software documentation. I see that SILE supports creating custom outputters, so maybe an HTML outputter could be created for that?

Related issue: #1337 (leet!) — this one is for developer documentation, but I think that manual could benefit from being available as a website too.

**Comment by alerque** [comment]
We're certainly considered this and from time to time it still crosses my mind, but there is a huge problem with the idea: the developer docs with Lua APIs are a little bit different, but as far as end user facing document commands and classes go the thing we are documenting is specifically designed for output to PDF / print. Having the documentation in PDF allows the techniques to de demonstrated inline. The alternative would be we need to generate HTML or some other markup and also generate a million illustration images that are snippets of PDF's converted to images so the examples can be seen. Even at that the result would not be nearly as informative as seeing the changes inline in the PDF.

It's not off the table entirely, but even if we were to write up a new outputter which mimicked the PDF output in some other markup environment, the results would inevitably be different and lead to confusion.

**Comment by nawordar** [comment]
I've just learned that TikZ  has an unofficial HTML documentation at https://tikz.dev. It is converted from LaTeX using [lwarp](https://ctan.org/pkg/lwarp). I haven't yet looked at how that package works, but the results look really good. Just saving it for the future reference, so maybe I will look into it later.

**Comment by hegjon** [comment]
Not sure if if it is possible to convert the PDF into a input format that Pandoc supports?
https://pandoc.org/MANUAL.html#general-options

### Specify input format. FORMAT can be:

- bibtex ([BibTeX](https://ctan.org/pkg/bibtex) bibliography)
- biblatex ([BibLaTeX](https://ctan.org/pkg/biblatex) bibliography)
- bits ([BITS](https://jats.nlm.nih.gov/extensions/bits/) XML, alias for jats)
- commonmark ([CommonMark](https://commonmark.org/) Markdown)
- commonmark_x ([CommonMark](https://commonmark.org/) Markdown with extensions)
- creole ([Creole 1.0](http://www.wikicreole.org/wiki/Creole1.0))
- csljson ([CSL JSON](https://citeproc-js.readthedocs.io/en/latest/csl-json/markup.html) bibliography)
- csv ([CSV](https://tools.ietf.org/html/rfc4180) table)
- tsv ([TSV](https://www.iana.org/assignments/media-types/text/tab-separated-values) table)
- docbook ([DocBook](https://docbook.org/))
- docx ([Word docx](https://en.wikipedia.org/wiki/Office_Open_XML))
- dokuwiki ([DokuWiki markup](https://www.dokuwiki.org/dokuwiki))
- endnotexml ([EndNote XML bibliography](https://support.clarivate.com/Endnote/s/article/EndNote-XML-Document-Type-Definition))
- epub ([EPUB](http://idpf.org/epub))
- fb2 ([FictionBook2](http://www.fictionbook.org/index.php/Eng:XML_Schema_Fictionbook_2.1) e-book)
- gfm ([GitHub-Flavored Markdown](https://help.github.com/articles/github-flavored-markdown/)), or the deprecated and less accurate markdown_github; use [markdown_github](https://pandoc.org/MANUAL.html#markdown-variants) only if you need extensions not supported in [gfm](https://pandoc.org/MANUAL.html#markdown-variants).
- haddock ([Haddock markup](https://www.haskell.org/haddock/doc/html/ch03s08.html))
- html ([HTML](https://www.w3.org/html/))
- ipynb ([Jupyter notebook](https://nbformat.readthedocs.io/en/latest/))
- jats ([JATS](https://jats.nlm.nih.gov/) XML)
- jira ([Jira](https://jira.atlassian.com/secure/WikiRendererHelpAction.jspa?section=all)/Confluence wiki markup)
- json (JSON version of native AST)
- latex ([LaTeX](https://www.latex-project.org/))
- markdown ([Pandoc’s Markdown](https://pandoc.org/MANUAL.html#pandocs-markdown))
- markdown_mmd ([MultiMarkdown](https://fletcherpenney.net/multimarkdown/))
- markdown_phpextra ([PHP Markdown Extra](https://michelf.ca/projects/php-markdown/extra/))
- markdown_strict (original unextended [Markdown](https://daringfireball.net/projects/markdown/))
- mediawiki ([MediaWiki markup](https://www.mediawiki.org/wiki/Help:Formatting))
- man ([roff man](https://man.cx/groff_man(7)))
- muse ([Muse](https://amusewiki.org/library/manual))
- native (native Haskell)
- odt ([ODT](https://en.wikipedia.org/wiki/OpenDocument))
- opml ([OPML](http://dev.opml.org/spec2.html))
- org ([Emacs Org mode](https://orgmode.org/))
- ris ([RIS](https://en.wikipedia.org/wiki/RIS_(file_format)) bibliography)
- rtf ([Rich Text Format](https://en.wikipedia.org/wiki/Rich_Text_Format))
- rst ([reStructuredText](https://docutils.sourceforge.io/docs/ref/rst/introduction.html))
- t2t ([txt2tags](https://txt2tags.org/))
- textile ([Textile](https://textile-lang.com/))
- tikiwiki ([TikiWiki markup](https://doc.tiki.org/Wiki-Syntax-Text#The_Markup_Language_Wiki-Syntax))
- twiki ([TWiki markup](https://twiki.org/cgi-bin/view/TWiki/TextFormattingRules))
- typst ([typst](https://typst.app/))
- vimwiki ([Vimwiki](https://vimwiki.github.io/))
- the path of a custom Lua reader, see [Custom readers and writers](https://pandoc.org/MANUAL.html#custom-readers-and-writers) below

**Comment by Omikhleia** [comment]
@hegjon 
> Not sure if if it is possible to convert the PDF into a input format that Pandoc supports

No, not really, because (as @alerque actually mentions above), the current manual in many places demonstrates things that result from processing, and would lose that in the conversion process from the source.

As mentioned, an HTML outputter could be created (I have a raw experimental one actually -- but there would be a lot more things needed to make it useful)...

Or, of couse, one could make a subset of the documentation without the actual demonstrations...

**Comment by alerque** [comment]
As @Omikheia says..

On the flip side, we *can* use code snippets to generate images to embed the demonstration output into a web version of the docs.

Any such attempt to retool should come *after* the more approachable step of generating API docs for Lua and Rust APIs. These would be much more technical and coding oriented (the expected arguments and variables) than design oriented and would be much easier to present in a web format while still keeping the "Manual" in a typeset format that much more readily demonstrates concepts instead of just talking about them.

**Comment by hegjon** [comment]
> No, not really, because (as @alerque actually mentions above), the current manual in many places demonstrates things that result from processing, and would lose that in the conversion process from the source.

I discovered the same when playing around with converting the manual.
